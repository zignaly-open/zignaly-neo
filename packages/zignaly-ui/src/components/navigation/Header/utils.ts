// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
