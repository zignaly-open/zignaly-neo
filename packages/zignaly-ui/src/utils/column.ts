import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
