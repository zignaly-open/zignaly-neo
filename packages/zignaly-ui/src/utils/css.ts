import { css } from "styled-components";

export const NiceScrollbar = css`
  &::-webkit-scrollbar {
    width: 6px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: rgba(255, 255, 255, 0.1);
    border-radius: 12px;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: rgba(255, 255, 255, 0.15);
  }
`;
