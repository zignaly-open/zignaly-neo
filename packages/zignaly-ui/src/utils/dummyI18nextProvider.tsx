import i18n from "i18next";
import React from "react";
import { I18nextProvider, initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
  lng: "en",
  fallbackLng: "en",
  ns: ["common"],
  defaultNS: "common",
  interpolation: {
    escapeValue: false,
  },
  resources: { en: { common: { xfuturum: "The technical partner you can rely on" } } },
});

export const storybookI18nextDecorator = (story: () => JSX.Element) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{story()}</I18nextProvider>
);

export const JestI18nextDecorator: React.FC<{ children: JSX.Element }> = ({ children }) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
);
