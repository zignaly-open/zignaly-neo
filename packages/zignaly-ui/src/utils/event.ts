export const changeEvent = (name: string | undefined, value: any) => {
  return {
    target: {
      value,
      name,
    },
    type: "change",
  };
};
