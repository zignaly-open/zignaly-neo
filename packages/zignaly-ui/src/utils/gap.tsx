import styled from "styled-components";

export const Gap = styled.div<{ gap: number }>`
  ${(props) => `
  padding: ${props.gap}px;
  `}
`;
