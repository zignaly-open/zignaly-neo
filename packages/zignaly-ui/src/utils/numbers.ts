export const sortByValue = (number1: string | number, number2: string | number) => {
  return +number1 - +number2;
};

export const trimZeros = (v: string | number) => {
  return v?.toString()?.replace(/(\.\d*?[1-9]+)0+$|\.0+$/g, "$1");
};

export const numberOfDecimals = (v: string | number) => {
  return (v?.toString()?.split(".")[1] ?? "").length;
};

export const formatCompactNumber = (value: number | string, precision = 8) =>
  new Intl.NumberFormat("en-US", {
    maximumFractionDigits: precision,
    notation: "compact",
    compactDisplay: "short",
  }).format(+value);
