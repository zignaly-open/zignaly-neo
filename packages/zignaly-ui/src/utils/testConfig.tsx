import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
