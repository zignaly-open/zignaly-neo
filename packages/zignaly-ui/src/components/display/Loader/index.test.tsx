import React from "react";
import { Loader, CenteredLoader } from "./index";
import { renderWithProvidersUi } from "../../../utils/testConfig";

describe("components/inputs/ZigButton", () => {
  describe("Loader rendering", () => {
    it("renders with default props", () => {
      const { container } = renderWithProvidersUi(<Loader />);

      const loader = container.querySelector(".simple-loader") as Element;
      expect(loader).toBeVisible();
      const loaderStyles = getComputedStyle(loader);
      expect(loader).toMatchSnapshot();
      expect(loaderStyles).toMatchSnapshot();
    });

    it("renders with custom width and height", () => {
      const { container } = renderWithProvidersUi(<Loader width={50} height={50} />);

      const loader = container.querySelector(".simple-loader") as Element;
      expect(loader).toBeVisible();
      const loaderStyles = getComputedStyle(loader);
      expect(loader).toMatchSnapshot();
      expect(loaderStyles).toMatchSnapshot();
    });
  });

  describe("Centered Loader rendering", () => {
    it("renders with default props", () => {
      const { container } = renderWithProvidersUi(<CenteredLoader />);

      const loader = container.querySelector(".centered-loader-wrapper") as Element;
      expect(loader).toBeVisible();
      const loaderStyles = getComputedStyle(loader);
      expect(loader).toMatchSnapshot();
      expect(loaderStyles).toMatchSnapshot();
    });
  });
});
