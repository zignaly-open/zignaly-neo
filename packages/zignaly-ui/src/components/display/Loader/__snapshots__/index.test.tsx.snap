// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/display/Loader should render with custom width and height without crashing 1`] = `
<div
  class="simple-loader"
  data-testid="oval-loading"
  style="display: flex; padding: 3px;"
>
  <svg
    aria-label="Loading..."
    data-testid="oval-svg"
    height="50px"
    stroke="#F3F4F6"
    viewBox="-20 -20 42 42"
    width="50px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fill-rule="evenodd"
    >
      <g
        stroke-width="2"
        transform="translate(1 1)"
      >
        <circle
          cx="0"
          cy="0"
          r="20"
          stroke="#F3F4F633"
          stroke-opacity=".5"
          stroke-width="2"
        />
        <path
          d="M20 0c0-9.94-8.06-20-20-20"
        >
          <animatetransform
            attributeName="transform"
            dur="1s"
            from="0 0 0"
            repeatCount="indefinite"
            to="360 0 0"
            type="rotate"
          />
        </path>
      </g>
    </g>
  </svg>
</div>
`;

exports[`components/display/Loader should render with custom width and height without crashing 2`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "visibility",
  "2": "padding",
  "_importants": {
    "display": "",
    "padding": "",
    "visibility": undefined,
  },
  "_length": 3,
  "_onChange": [Function],
  "_values": {
    "display": "flex",
    "padding": "3px",
    "padding-bottom": "3px",
    "padding-left": "3px",
    "padding-right": "3px",
    "padding-top": "3px",
    "visibility": "visible",
  },
}
`;

exports[`components/display/Loader should render with default props without crashing 1`] = `
<div
  class="simple-loader"
  data-testid="oval-loading"
  style="display: flex; padding: 3px;"
>
  <svg
    aria-label="Loading..."
    data-testid="oval-svg"
    height="40px"
    stroke="#F3F4F6"
    viewBox="-20 -20 42 42"
    width="40px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fill-rule="evenodd"
    >
      <g
        stroke-width="2"
        transform="translate(1 1)"
      >
        <circle
          cx="0"
          cy="0"
          r="20"
          stroke="#F3F4F633"
          stroke-opacity=".5"
          stroke-width="2"
        />
        <path
          d="M20 0c0-9.94-8.06-20-20-20"
        >
          <animatetransform
            attributeName="transform"
            dur="1s"
            from="0 0 0"
            repeatCount="indefinite"
            to="360 0 0"
            type="rotate"
          />
        </path>
      </g>
    </g>
  </svg>
</div>
`;

exports[`components/display/Loader should render with default props without crashing 2`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "visibility",
  "2": "padding",
  "_importants": {
    "display": "",
    "padding": "",
    "visibility": undefined,
  },
  "_length": 3,
  "_onChange": [Function],
  "_values": {
    "display": "flex",
    "padding": "3px",
    "padding-bottom": "3px",
    "padding-left": "3px",
    "padding-right": "3px",
    "padding-top": "3px",
    "visibility": "visible",
  },
}
`;

exports[`components/display/Loader should render with default props without crashing 3`] = `
<div
  class="centered-loader-wrapper MuiBox-root css-1lhu1y1"
>
  <div
    class="simple-loader"
    data-testid="oval-loading"
    style="display: flex; padding: 3px;"
  >
    <svg
      aria-label="Loading..."
      data-testid="oval-svg"
      height="40px"
      stroke="#F3F4F6"
      viewBox="-20 -20 42 42"
      width="40px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <g
          stroke-width="2"
          transform="translate(1 1)"
        >
          <circle
            cx="0"
            cy="0"
            r="20"
            stroke="#F3F4F633"
            stroke-opacity=".5"
            stroke-width="2"
          />
          <path
            d="M20 0c0-9.94-8.06-20-20-20"
          >
            <animatetransform
              attributeName="transform"
              dur="1s"
              from="0 0 0"
              repeatCount="indefinite"
              to="360 0 0"
              type="rotate"
            />
          </path>
        </g>
      </g>
    </svg>
  </div>
</div>
`;

exports[`components/display/Loader should render with default props without crashing 4`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "padding",
  "2": "align-items",
  "3": "justify-content",
  "4": "visibility",
  "_importants": {
    "align-items": "",
    "display": "",
    "justify-content": "",
    "padding": "",
    "visibility": undefined,
  },
  "_length": 5,
  "_onChange": [Function],
  "_values": {
    "align-items": "center",
    "display": "flex",
    "justify-content": "center",
    "padding": "8em 0px",
    "padding-bottom": "8em",
    "padding-left": "0px",
    "padding-right": "0px",
    "padding-top": "8em",
    "visibility": "visible",
  },
}
`;
